doc/modules/ROOT/pages/index.adoc

```diff
@@ -1,46 +1,235 @@
+
+[source, shell]
+----
+$ export AWS_ACCESS_KEY_ID="XXXXXXXXXXXX"
+$ export AWS_SECRET_ACCESS_KEY="XXXXXXXXXXXXXXXX"
+----
+
+You can find other ways of providing credentials at https://registry.terraform.io/providers/hashicorp/aws/latest/docs#authentication[Terraform AWS authentication].
+
+The account you use in Terraform should have the following permissions assigned to it:
+ 
+----
+"iam:CreateRole",
+"iam:DeleteRole",
+"iam:CreateInstanceProfile",
+"iam:DeleteInstanceProfile",
+"iam:DeleteRolePolicy",
+"iam:AddRoleToInstanceProfile",
+"iam:RemoveRoleFromInstanceProfile",
+"iam:PutRolePolicy",
+"iam:PassRole",
+"iam:GetRole",
+"iam:GetInstanceProfile",
+"iam:GetPolicy",
+"iam:GetRolePolicy"
+"ec2:*"
+----
+
+The `iam` permissions are necessary to create and delete a role using Terraform in `aws/main.tf`. Created role `aws_iam_role.discovery_role` will be used by Hazelcast instances to find each other.
+--
+
+Azure::
++
+--
+
+If you are using a user account, you can login with https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest#install[Azure CLI]. Run the following command to authenticate. Terraform will be able to detect your account.
+
+[source, shell]
+----
+$ az login
+----
+
+If you have multiple subscriptions or tenants you can choose one by adding following lines to `azure/main.tf`.
+[source, shell]
+----
+provider "azurerm" {
+  version = "=2.23.0"
+  features {}
+
+  subscription_id = "00000000-0000-0000-0000-000000000000"
+  tenant_id       = "11111111-1111-1111-1111-111111111111"
+
+}
+
+If you want to authenticate through service principals, please refer to https://www.terraform.io/docs/providers/azurerm/index.html#authenticating-to-azure[Authenticating to Azure].
+----
 
-////
-Write the steps in your guide here by making each step a different section
-////
+
+The account or service principal you use should have the role `Owner` assigned to it.
+
+--
+
+
+GCP::
++
+--
+
+You can use service accounts to authenticate Terraform. Get a service account key file, you can create key files on https://console.cloud.google.com/apis/credentials/serviceaccountkey[Google Console]. After you have created a key file, put its path  in `gcp/main.tf` as follows.
+
+[source,terraform]
+----
+provider "google" {
+  version = "3.5.0"
+
+  credentials = file("KEY-FILE-PATH/YOUR-KEY-FILE.json")
+  batching {
+    enable_batching = "false"
+  }
+  project = var.project_id
+  region  = var.region
+  zone    = var.zone
+}
+
+----
+The service account you use should have the following roles assigned to it:
+
+----
+Compute Admin
+Role Administrator
+Security Admin
+Service Account Admin
+Service Account User 
+----
+--
+====
+== Configuring Terraform for Connection
+
+Now Terraform has access to your credentials, we need to supply some variables to configure Terraform to be able to create resources correctly.
```

Comment:
```suggestion
Now that Terraform has access to your credentials, we need to supply some variables to configure Terraform so that resources can be created correctly.
```